internal/controller/serverset/bootvolume_controller_test.go

```diff
@@ -0,0 +1,104 @@
+package serverset
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"testing"
+
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/client/interceptor"
+	_ "sigs.k8s.io/controller-runtime/pkg/envtest"
+
+	"github.com/ionos-cloud/crossplane-provider-ionoscloud/apis/compute/v1alpha1"
+	computev1alpha1 "github.com/ionos-cloud/crossplane-provider-ionoscloud/apis/compute/v1alpha1"
+)
+
+func fakeKubeClient(functions interceptor.Funcs) client.WithWatch {
+	scheme, _ := computev1alpha1.SchemeBuilder.Build()
+	return fake.NewClientBuilder().WithScheme(scheme).WithInterceptorFuncs(functions).Build()
+}
+
+func getVolumePopulateStatusTest(ctx context.Context, client client.WithWatch, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
+	vol := obj.(*v1alpha1.Volume)
+	vol.Status.AtProvider.State = "AVAILABLE"
+	vol.Status.AtProvider.VolumeID = "uuid"
+	return nil
+}
+
+func createVolumeReturnsErrorTest(ctx context.Context, client client.WithWatch, obj client.Object,
+	opts ...client.CreateOption) error {
+	return errors.New("something went wrong")
+}
+
+func Test_kubeBootVolumeController_Create(t *testing.T) {
+	type fields struct {
+		kube client.Client
+		log  logging.Logger
+	}
+	type args struct {
+		ctx          context.Context
+		cr           *v1alpha1.ServerSet
+		replicaIndex int
+		version      int
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    v1alpha1.Volume
+		wantErr bool
+	}{
+		{
+			name: "expect success",
+			fields: fields{
+				kube: fakeKubeClient(interceptor.Funcs{Get: getVolumePopulateStatusTest}),
```

Comment:
I think `test` can be removed from the method name: `getVolumeWithStatusPopulated`

Reply:
I'd like to keep them just so people know when looking at functions from this package that these are test functions.

Reply:
well they are in a *_test*.go file and they are private, so you cannot use them in other modules

Reply:
in other modules no, but if someone searches code it's easier for them to figure out from function name. will change them.

Reply:
done